packages/fxa-settings/src/pages/ResetPassword/ResetPasswordWithRecoveryKeyVerified/container.tsx

```diff
@@ -45,15 +45,24 @@ const ResetPasswordWithRecoveryKeyVerifiedContainer = ({
 
   const account = useAccount();
   const { uid, sessionToken } = currentAccount() || {};
-  const { keyFetchToken, unwrapBKey } =
-    sensitiveDataClient.getDataType(SensitiveData.Key.AccountReset) || {};
-  const newRecoveryKeyData = sensitiveDataClient.getData('newRecoveryKeyData');
-  const newRecoveryKey = formatRecoveryKey(newRecoveryKeyData.buffer);
-
   const updateRecoveryKeyHint = useCallback(
     async (hint: string) => account.updateRecoveryKeyHint(hint),
     [account]
   );
+  const { keyFetchToken, unwrapBKey } =
+    sensitiveDataClient.getDataType(SensitiveData.Key.AccountReset) || {};
+  const recoveryKey = sensitiveDataClient.getDataType(
+    SensitiveData.Key.NewRecoveryKey
+  )?.recoveryKey;
+
+  if (!recoveryKey) {
+    // If we get here, that means a password reset was completed (with a verified account).
+    // Along the way, we have lost a copy of the recovery key in memory if the page was unloaded.
+    // This is fine - we navigate to the confirmed page and carry on.
+    navigateWithQuery('/reset_password_confirmed');
+    return;
+  }
+  const newRecoveryKey = formatRecoveryKey(recoveryKey);
```

Comment:
Locally, I'm seeing a type error for `recoveryKey` here and for `recoveryKeyBuffer` in the `formatRecoveryKey` function. Do you see the same? We may need to pass in `recoveryKey.buffer` as is done in other components that use the formatRecoveryKey function (but I haven't investigated fully)

Reply:
Oops, I had this fixed locally. You're right that it was supposed to be `recoveryKey.buffer`, but it alludes me why my build did not fail on such an obvious error..